reference/activity_manager/project_completeness.rst

```diff
@@ -0,0 +1,110 @@
+
+This first method seems not to be the chosen one. Let's recap which data we need:
+ - For each locale,
+ - For each channel,
+ - For each user,
+ - Does the product
+ - have at least an attribute filled in?
+ - or is it complete?
+
+Oh. Can you see this?
+
+Doing a table to gather all data we need to compute the completeness seems to be a good idea but we need some
+adjustments. It can't be scalable if we do this for each user.
+
+.. _format: format.html
+
+So let's remove users from this table. And now, how to retrieve the completeness from a user? In the project format_,
+we have product filters. From those filters it's easy to know which user is impacted by the products selection. To win
+some times, we pre-calculate this data linking users impacted to project at its creation. But it's not scalable too. If
+we add a user, its completeness could not be calculated because he is not linked to project yet. Problem is resolved
+linking user groups to project and not directly to users.
+
+Some data are still missing. User groups permissions are not simply about products but about attribute groups.
+
+Here is the final structure we needed (only data that are relevant for the demonstration are shown):
+ - Pre-processing table:
+  - For each locale,
+  - For each channel,
+  - Does the attribute group,
+  - in the product,
+  - have at least an attribute filled in?
+  - or is it complete?
+
+ - Project model:
+  - locale
+  - channel
+  - product filters
+  - products (*)
+  - user groups (*)
+
+.. _calculation step: calculation_step.html
+
+(*) As these fields are only used for the completeness calculation purpose and they have no sense from a business point
+of view, they are not mapped to doctrine. Products field is a n - n table that helps to join project to pre-processing.
+User groups field is a n - n table that helps to know who is impacted by the products selection. These table are filled
+thanks to the `calculation step`_.
+
+By cross-checking pre-processing table, project table, category accesses table, attribute group accesses table we are
+able to calculate those three little numbers that are the number of products not started to enrich, in progress and done
+in terms of a user and a products selection.
+
+In ORM, joining Project to Category access walk through:
+
+project - project_products - product_category - category - category_access
+
+Problem comes with ODM storage where `product_category` is transferred in MongoDB. Joining all these tables in MySQL is
+not possible anymore. To avoid to do a lot of extra queries in MongoDB to fetch `product_category`, we added
+`PimEnterprise\Component\ActivityManager\Job\ProjectCalculation\CalculationStep\LinkProductAndCategoryStep` `calculation
+step`_. This step allow us to re-create this link table in MySQL and be able to calculate all completeness with MySQL
+even in ODM.
+
+This is the end
+_______________
+
+As the first mentioned method, drawback is that the completeness is not up to date.
```

Comment:
Still the drawback mentioned in the first method remains: the completeness is not up to date.